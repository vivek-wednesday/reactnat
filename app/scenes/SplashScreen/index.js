import React from 'react';
import styled from 'styled-components/native';
import AppContainer from '@atoms/Container';
import { colors, images } from '@themes';

const Container = styled(AppContainer)`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${colors.primary};
`;

const Logo = styled.Image`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 200px;
`;

const SplashScreen = () => (
  <Container testID="splash-screen">
    <Logo source={images.wednesdayLogo} resizeMode="contain" />
  </Container>
);

export default SplashScreen;
