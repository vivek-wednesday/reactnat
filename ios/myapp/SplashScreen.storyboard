<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="17156" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" launchScreen="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="EXPO-VIEWCONTROLLER-1">
    <device id="retina5_5" orientation="portrait" appearance="light"/>
    <dependencies>
        <deployment identifier="iOS"/>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="17125"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <scenes>
        <!--View Controller-->
        <scene sceneID="EXPO-SCENE-1">
            <objects>
                <viewController storyboardIdentifier="SplashScreenViewController" id="EXPO-VIEWCONTROLLER-1" sceneMemberID="viewController">
                    <view key="view" userInteractionEnabled="NO" contentMode="scaleToFill" insetsLayoutMarginsFromSafeArea="NO" id="EXPO-ContainerView" userLabel="ContainerView">
                        <rect key="frame" x="0.0" y="0.0" width="414" height="736"/>
                        <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                        <subviews>
                            <imageView userInteractionEnabled="NO" contentMode="scaleAspectFill" horizontalHuggingPriority="251" verticalHuggingPriority="251" insetsLayoutMarginsFromSafeArea="NO" image="SplashScreenBackground" translatesAutoresizingMaskIntoConstraints="NO" id="EXPO-SplashScreenBackground" userLabel="SplashScreenBackground">
                                <rect key="frame" x="0.0" y="0.0" width="414" height="736"/>
                            </imageView>
                            <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFit" horizontalHuggingPriority="251" verticalHuggingPriority="251" image="LaunchScreen" translatesAutoresizingMaskIntoConstraints="NO" id="lZ1-ge-8Re">
                                <rect key="frame" x="20" y="0.0" width="374" height="736"/>
                            </imageView>
                        </subviews>
                        <constraints>
                            <constraint firstItem="EXPO-SplashScreenBackground" firstAttribute="top" secondItem="EXPO-ContainerView" secondAttribute="top" id="1gX-mQ-vu6"/>
                            <constraint firstItem="EXPO-SplashScreenBackground" firstAttribute="leading" secondItem="EXPO-ContainerView" secondAttribute="leading" id="6tX-OG-Sck"/>
                            <constraint firstAttribute="bottomMargin" secondItem="lZ1-ge-8Re" secondAttribute="bottom" id="9tM-rv-eq6"/>
                            <constraint firstItem="EXPO-SplashScreenBackground" firstAttribute="trailing" secondItem="EXPO-ContainerView" secondAttribute="trailing" id="ABX-8g-7v4"/>
                            <constraint firstItem="lZ1-ge-8Re" firstAttribute="top" secondItem="EXPO-ContainerView" secondAttribute="topMargin" id="JcL-lT-fpO"/>
                            <constraint firstAttribute="trailingMargin" secondItem="lZ1-ge-8Re" secondAttribute="trailing" id="hRJ-pH-6Yz"/>
                            <constraint firstItem="EXPO-SplashScreenBackground" firstAttribute="bottom" secondItem="EXPO-ContainerView" secondAttribute="bottom" id="jkI-2V-eW5"/>
                            <constraint firstItem="lZ1-ge-8Re" firstAttribute="leading" secondItem="EXPO-ContainerView" secondAttribute="leadingMargin" id="zQ7-SQ-kKy"/>
                        </constraints>
                    </view>
                </viewController>
                <placeholder placeholderIdentifier="IBFirstResponder" id="EXPO-PLACEHOLDER-1" userLabel="First Responder" sceneMemberID="firstResponder"/>
            </objects>
            <point key="canvasLocation" x="140.625" y="129.4921875"/>
        </scene>
    </scenes>
    <resources>
        <image name="LaunchScreen" width="512" height="512"/>
        <image name="SplashScreenBackground" width="1" height="1"/>
    </resources>
</document>
