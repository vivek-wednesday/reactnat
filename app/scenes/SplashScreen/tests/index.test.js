/**
 *
 * Tests for SplashScreen
 *
 */
import React from 'react';
import { renderWithIntl } from 'app/utils/testUtils';
import SplashScreen from '../index';

describe('<SplashScreen />', () => {
  it('should render and match the snapshot', () => {
    const baseElement = renderWithIntl(<SplashScreen />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 splash screen', () => {
    const { getAllByTestId } = renderWithIntl(<SplashScreen />);
    expect(getAllByTestId('splash-screen').length).toBe(1);
  });
});
