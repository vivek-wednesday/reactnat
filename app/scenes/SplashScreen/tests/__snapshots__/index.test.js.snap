// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashScreen /> should render and match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fcedda",
        "display": "flex",
        "justifyContent": "center",
      },
    ]
  }
  testID="splash-screen"
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../app/assets/images/wednesday-logo.png",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "width": 200,
        },
      ]
    }
  />
</View>
`;
